### PR TITLE
feature/cart

### PR DESCRIPTION
**Pull request by Aidai0404**

### CODE REVIEW

**Review by Izat150487 — COMMENTED** [review]
viewed✅

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("api/cart")
+@RequiredArgsConstructor
+public class BasketController {
+
+    private final BasketService basketService;
+
+    @PostMapping("save-product-to-cart/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:23]: Endpoint naming convention

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    public ProductResponse saveProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return basketService.saveProductIntoBasket(id, productRequest);
+    }
+
+    @PutMapping("remove-product-from-cart/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:28]: Endpoint naming convention

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    public SimpleResponse deleteProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return basketService.removeProductFromBasket(id, productRequest);
+    }
+
+
+    @PutMapping("remove-all-product/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: Endpoint naming convention

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public SimpleResponse deleteAllProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest){
+        return basketService.removeAllProductFromBasket(id,productRequest);
+    }
+
+    @GetMapping("products-of-cart/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:39]: Endpoint naming convention

**Review by i-Coderr — COMMENTED** [review]
Pls provide the Endpoint naming convention properly

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("api/cart")
+@RequiredArgsConstructor
+public class BasketController {
+
+    private final BasketService basketService;
+
+    @PostMapping("save-product/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:23]: Rewrite to products/{id}

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    public ProductResponse saveProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return basketService.saveProductIntoBasket(id, productRequest);
+    }
+
+    @PutMapping("remove-product/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:28]: Rewrite to products/{id}

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    public SimpleResponse deleteProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return basketService.removeProductFromBasket(id, productRequest);
+    }
+
+
+    @PutMapping("remove-products/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: Rewrite to all-products/{id}

Comment (i-Coderr) [reply src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: Here id is which id?

**Review by i-Coderr — COMMENTED** [review]
Commented at 12:13 am

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java (hunk 7 above)
Comment (Aidai0404) [reply src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: when i wrote remove-all=products, you said i should name it in other way

Comment (Aidai0404) [reply src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: basket's id

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("api/cart")
+@RequiredArgsConstructor
+public class BasketController {
+
+    private final BasketService basketService;
+
+    @PostMapping("save-products/{id}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:23]: Endpoint naming convention (remove save)

File: src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        return basketService.saveProductIntoBasket(id, productRequest);
+    }
+
+    @PutMapping("remove-products/{id}")
+    public SimpleResponse deleteProduct(@PathVariable("id") Long id, @RequestBody ProductRequest productRequest) {
+        return basketService.removeProductFromBasket(id, productRequest);
+    }
+
+
+    @PutMapping("remove-all-products/{basketId}")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/BasketController.java:34]: Same

**Review by i-Coderr — COMMENTED** [review]
Pls add swagger

**Review by i-Coderr — APPROVED** [review]
Alright

**Pull request merged**